src/test/scala/chiselTests/StrongEnum.scala

```diff
@@ -367,62 +367,198 @@ class StrongEnumSpec extends ChiselFlatSpec {
     "foo(OtherEnum.otherEnum)" shouldNot compile
   }
 
+  it should "prevent enums from being declared without names" in {
+    "object UnnamedEnum extends ChiselEnum { Value }" shouldNot compile
+  }
+
   "StrongEnum FSM" should "work" in {
     assertTesterPasses(new StrongEnumFSMTester)
   }
 }
 
-class StrongEnumAnnotationSpec extends ChiselFlatSpec {
+class StrongEnumAnnotator extends Module {
+  import EnumExample._
+
+  val io = IO(new Bundle{
+    val in = Input(EnumExample())
+    val out = Output(EnumExample())
+    val other = Output(OtherEnum())
+  })
+
+  class Bund extends Bundle {
+    val field = EnumExample()
+    val other = OtherEnum()
+    val vec = Vec(5, EnumExample())
+    val inner_bundle1 = new Bundle {
+      val x = UInt(4.W)
+      val y = Vec(3, UInt())
+      val e = EnumExample()
+      val v = Vec(3, EnumExample())
+    }
+    val inner_bundle2 = new Bundle {}
+    val inner_bundle3 = new Bundle {
+      val x = Bool()
+    }
+    val inner_bundle4 = new Bundle {
+      val inner_inner_bundle = new Bundle {}
+    }
+
+    def init(): Unit = {
+      field := e0
+      other := OtherEnum.otherEnum
+      vec.foreach(_ := e0)
+      inner_bundle1.x := 0.U
+      inner_bundle1.y.foreach(_ := 0.U)
+      inner_bundle1.e := e0
+      inner_bundle1.v.foreach(_ := e0)
+      inner_bundle3.x := false.B
+    }
+  }
+
+  val simple = Wire(EnumExample())
+  val vec = VecInit(e0, e1, e2)
+  val vec_of_vecs = VecInit(VecInit(e0, e1), VecInit(e100, e101))
+
+  val bund = Wire(new Bund())
+  val vec_of_bundles = Wire(Vec(5, new Bund()))
+
+  io.out := e101
+  io.other := OtherEnum.otherEnum
+  simple := e100
+  bund.init()
+  vec_of_bundles.foreach(_.init())
+
+  // All variables starting with "ignored" must not be annotated
+  val cycle = RegInit(0.U)
+  cycle := cycle + 1.U
+
+  val ignore = EnumExample()
```

Comment:
What is this ignoring thing? Can you add a little more comment about what is being tested/demonstrated here

Reply:
I just got rid of the _ignore_ tests. On further thought, I don't think they're needed anyway.